sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp

```diff
@@ -0,0 +1,860 @@
+{
+  if (samples.size () != sample_size_)
+  {
+    PCL_ERROR ("[pcl::SampleConsensusModelEllipse3D::isSampleGood] Wrong number of samples (is %lu, should be %lu)!\n", samples.size (), sample_size_);
+    return (false);
+  }
+
+  // Use three points out of the 6 samples
+  Eigen::Vector3d p0 ((*input_)[samples[0]].x, (*input_)[samples[0]].y, (*input_)[samples[0]].z);
+  Eigen::Vector3d p1 ((*input_)[samples[1]].x, (*input_)[samples[1]].y, (*input_)[samples[1]].z);
+  Eigen::Vector3d p2 ((*input_)[samples[2]].x, (*input_)[samples[2]].y, (*input_)[samples[2]].z);
+
+  // Check if the squared norm of the cross-product is non-zero, otherwise
+  // common_helper_vec, which plays an important role in computeModelCoefficients,
+  // would likely be ill-formed.
+  if ((p1 - p0).cross(p1 - p2).squaredNorm() < Eigen::NumTraits<float>::dummy_precision ())
+  {
+    PCL_ERROR ("[pcl::SampleConsensusModelEllipse3D::isSampleGood] Sample points too similar or collinear!\n");
+    return (false);
+  }
+
+  return (true);
+}
+
+//////////////////////////////////////////////////////////////////////////
+template <typename PointT> bool
+pcl::SampleConsensusModelEllipse3D<PointT>::computeModelCoefficients (const Indices &samples, Eigen::VectorXf &model_coefficients) const
+{
+  // Uses 6 samples
+  if (samples.size () != sample_size_)
+  {
+    PCL_ERROR ("[pcl::SampleConsensusModelEllipse3D::computeModelCoefficients] Invalid set of samples given (%lu)!\n", samples.size ());
+    return (false);
+  }
+
+  model_coefficients.resize (model_size_); // 11 coefs
+
+  Eigen::Vector3f p0((*input_)[samples[0]].x, (*input_)[samples[0]].y, (*input_)[samples[0]].z);
+  Eigen::Vector3f p1((*input_)[samples[1]].x, (*input_)[samples[1]].y, (*input_)[samples[1]].z);
+  Eigen::Vector3f p2((*input_)[samples[2]].x, (*input_)[samples[2]].y, (*input_)[samples[2]].z);
+  Eigen::Vector3f p3((*input_)[samples[3]].x, (*input_)[samples[3]].y, (*input_)[samples[3]].z);
+  Eigen::Vector3f p4((*input_)[samples[4]].x, (*input_)[samples[4]].y, (*input_)[samples[4]].z);
+  Eigen::Vector3f p5((*input_)[samples[5]].x, (*input_)[samples[5]].y, (*input_)[samples[5]].z);
+
+  Eigen::Vector3f common_helper_vec = (p1 - p0).cross(p1 - p2);
+  Eigen::Vector3f ellipse_normal = common_helper_vec.normalized();
+
+  // The same check is implemented in isSampleGood, so be sure to look there too
+  // if you find the need to change something here.
+  if (common_helper_vec.squaredNorm() < Eigen::NumTraits<float>::dummy_precision ())
+  {
+    PCL_ERROR ("[pcl::SampleConsensusModelEllipse3D::computeModelCoefficients] Sample points too similar or collinear!\n");
+    return (false);
+  }
+
+  // Definition of the local reference frame of the ellipse
+  Eigen::Vector3f x_axis = (p1 - p0).normalized();
+  Eigen::Vector3f z_axis = ellipse_normal.normalized();
+  Eigen::Vector3f y_axis = z_axis.cross(x_axis).normalized();
+  
+  // Create the transposed rotation matrix
+  Eigen::Matrix3f Rot;
```

Comment:
We can have const matrix like this:

  const Eigen::Matrix3f Rot = (Eigen::Matrix3f(3,3)
    << x_axis(0), y_axis(0), z_axis(0),
       x_axis(1), y_axis(1), z_axis(1),
       x_axis(2), y_axis(2), z_axis(2))
       .finished();

Reply:
I was not aware of this .finished() method, and most of the Eigen vecs/mats were not compiling as const. Thanks a lot! I'll change everything accordingly asap.